### PR TITLE
ci: Replace deprecated `hubble observe -o json` with `-o jsonpb`

### PR DESCRIPTION
**Pull request by gandro**
This commit replaces the use of the deprecated `-o json` flag on Hubble observe with `-o jsonpb`. Future versions of Hubble will make `-o json` an alias to `-o jsonpb`, so this commit should be future-proof.

The new output wraps each flow in a `GetFlowsResponse`, meaning the old object is now accessible in the `.flow` attribute. All jsonpath queries in the code have been changed to reflect this change.

Notably, some parts of CI (namely the `hubble-flows-*.json` files used for troubleshooting CI flakes) already used `-o jsonpb` and thus this commit should not cause any change in usual the CI troubleshooting workflow.

### CODE REVIEW

File: test/k8s/hubble.go
Comment (kaworu) [inline test/k8s/hubble.go:285]: nit: Since we're dereferencing anyway we could
```suggestion
			Expect(flows[0].Flow.Destination.Identity).To(BeNumerically(">=", identity.MinimalNumericIdentity))
```
We'll get the same error in the failure case, and in the context of tests accessing the struct members directly avoids hiding where the `nil` value is in the chain, if that make sense.

**Review by kaworu — APPROVED** [review]
Thanks @gandro :pray:

File: test/k8s/hubble.go
Comment (gandro) [reply test/k8s/hubble.go:285]: `.Flow` doesn't work here unfortunately, since `GetFlow()` does some type casting to determine the `oneof` response:

https://github.com/cilium/cilium/blob/1203d9e841b6cb10373bb88a0196f1d62fe78466/api/v1/observer/observer.pb.go#L686-L691

Comment (kaworu) [reply test/k8s/hubble.go:285]: Ah yes thank you I forgot about that :+1:

**Comment by gandro** [comment]
ci-eks and ci-aks failiure are unrelated. The affected code here is only active in Jenkins CI.

Once the remaining code owners approve, this is ready to merge.

I'm also marking for backport to v1.13 because we'll likely want to ship the new version of the Hubble CLI with Cilium 1.13.

**Review by christarazi — APPROVED** [review]

**Pull request merged**
